### PR TITLE
Rename “compilation member” as “definition”

### PR DESCRIPTION
**Pull request by paulhuggett**
(See also [pstore PR #77](https://github.com/SNSystems/pstore/pull/77) which is a prerequisite for this PR.)

A long time ago we decided to rename the repo’s “compilation member” as ”definition” to better represent its role in the schema. It’s documented this way on the prepo glossary page. Since then I’ve tried to ensure that new code used the new terminology, but there never seemed to be a good time to change the old. This has result in a confusing mishmash of terms that really needs to be resolved.

### CODE REVIEW

**Review by MaggieYingYi — COMMENTED** [review]
Some comments use the compilation member:

File Name: RepoObjectWriter.cpp
Line numbers: 83, 702 and 1015.

File Name: RepoFragments.cpp
Line number: 82

In the rld project:

File Name: Symbol.h
Line number: 192

File Name: TestXfxResolver.cpp
Line number: 170

File Name: rld.cpp
Line number: 224, 244

File: llvm/include/llvm/IR/RepoDefinition.h [hunk 1]
```diff
@@ -410,10 +410,9 @@ class RepoDefinition : public MDNode {
     return MD->getMetadataID() == RepoDefinitionKind;
   }
 
-  // A pointer to the corresponding compilation member in the repository. It is
-  // updated when the compilation member is generated by the object writer.
-  const pstore::repo::compilation_member *CorrespondingCompilationMember =
-      nullptr;
+  // A pointer to the corresponding definition in the repository. It is
+  // updated when the definition is generated by the object writer.
```
[the remark this answers is not in the document]
Comment (paulhuggett) [reply llvm/include/llvm/IR/RepoDefinition.h:414]: Everything in the change went through clang-format before being committed.

File: llvm/lib/MC/RepoObjectWriter.cpp [hunk 2]
```diff
@@ -561,24 +561,23 @@ void RepoObjectWriter::writeSectionData(ContentsType &Fragments,
...
   for (auto const &LD : LinkedDefinitions) {
     auto &D = Fragments[LD.first].LinkedDefinitions;
     for (auto const &TN : LD.second) {
-      assert(TN->CorrespondingCompilationMember >= Definitions.data() &&
-             TN->CorrespondingCompilationMember <= &Definitions.back() &&
-             "The corresponding compilation_member must lie within the "
+      assert(TN->CorrespondingDefinition >= Definitions.data() &&
+             TN->CorrespondingDefinition <= &Definitions.back() &&
+             "The corresponding definition must lie within the "
              "Definitions container.");
```
[the remark this answers is not in the document]
Comment (paulhuggett) [reply llvm/lib/MC/RepoObjectWriter.cpp:571]: As above.

**Comment by paulhuggett** [comment]
> Some comments use the compilation member:

I would have sworn that I’d searched for all of the various spellings and capitalisations in both projects. Guess I missed one… af0d2b1bb05b7c185b3a875e8ef6dfa6ce9cae99.

File: llvm/include/llvm/IR/RepoDefinition.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (MaggieYingYi) [reply llvm/include/llvm/IR/RepoDefinition.h:414]: For the comment, I moved the line 414 to the end of line 413 and then used the clang-format, You will see the format changes.

File: llvm/lib/MC/RepoObjectWriter.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (MaggieYingYi) [reply llvm/lib/MC/RepoObjectWriter.cpp:569]: Same as above. I put the "The corresponding definition must lie within the Definitions container." in the same line and used the clang-format, the format changed.

**Review by MaggieYingYi — APPROVED** [review]
LGTM!

File: llvm/include/llvm/IR/RepoDefinition.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (paulhuggett) [reply llvm/include/llvm/IR/RepoDefinition.h:414]: Your version of clang-format appears to be behaving differently to mine.

File: llvm/lib/MC/RepoObjectWriter.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (paulhuggett) [reply llvm/lib/MC/RepoObjectWriter.cpp:569]: As above.

**Comment by paulhuggett** [comment]
> LGTM!

Thanks. I can’t of course commit this work until the matching [pstore API change](https://github.com/SNSystems/pstore/pull/77) has been committed.

**Pull request merged**